src/bin-utils/parser.js

```diff
@@ -129,7 +136,9 @@ function parse(rawArgv) {
       log.info(specificHelpScript(psConfig, specifiedScripts[1]))
       return true
     } else if (commandIsHelp || noScriptSpecifiedAndNoDefault) {
-      parser.showHelp('log')
+      if (helpStyle === 'all' || helpStyle === 'undefined') {
```

Comment:
I would change this to: `helpStyle === 'all' || !helpStyle`

Unless someone explicitly sets `helpStyle` to the string of `'undefined'` then that second branch will never be true.

Reply:
@kentcdodds okay, so the undefined check was because of the value coming back from the tests. Where I think I might be getting confused and what might be happening is that, I'm expecting the [baseOptions](https://github.com/iamkether/nps/blob/508648d62871e490e98495b1768300a02b0dacc9/src/bin-utils/parser.js#L32-L61) to be setting a default _everytime_ the script is run. However, I don't think that's happening when the tests are run, so I'm getting an undefined value for my `helpStyle` variable. 

For example, if I change it back to this:
```js
if (helpStyle === 'all') {
    parser.showHelp('log')
}
```
and change the `helpStyle` var to have a default as I had it before:
```js
const helpStyle = String(psConfig.options['help-style'] || 'all')
```

I will get 100% function/statement/line coverage again, but still 97.67% on the branch stats for that same `if (helpStyle === 'all')` line. Now, I've read up some more on "branches" and I understand that Jest is calculating multiple different potential branches from this line:
- helpStyle === 'all'
- helpStyle === 'scripts'
- helpStyle === 'basic'
- helpStyle === undefined (in the case of aforementioned defaults during tests not being set)

But the way that I've written this is that `scripts` and `basic` values are not relevant at this point of the code, because they're analyzed inside of the `scriptObjectToChalk()` function from `index.js`. As far as I can tell, it seems to me like I'm going to need to do a bunch of refactoring to move `showHelp` into `index.js` and export it to be used in `parser.js`, because it's not currently exposed as a publically accessible method of the `parse` object exported from `parser.js`. 

It _ultimately_ makes sense for all of the "help"-related functions/screen output to be within the same file, but I want to check if you see another way before I dive into this because it's going to be a decent-sized refactoring.